chat-app/src/App.js

```diff
@@ -0,0 +1,177 @@
+
+import { useEffect, useState } from 'react';
+import './App.css';
+import { ChatRoom } from './ChatRoom';
+
+export function App() {
+  const [username, setUsername] = useState('');
+  const [roomId, setRoomId] = useState('');
+  const [availableRooms, setAvailableRooms] = useState([]);
+  const [socket, setSocket] = useState(null);
+  const [connectionsCount, setConnectionsCount] = useState(0);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [roomName, setRoomName] = useState('');
+
+  const createRoom = async () => {
+    if (!roomName || roomName.trim() === '') {
+      setErrorMessage('Room name cannot be empty');
+      return;
+    }
+
+    try {
+      const response = await fetch('http://localhost:3005/rooms', {
+        method: 'POST',
+        headers: { 'Content-Type': 'application/json' },
+        body: JSON.stringify({ roomName }),
+      });
+
+      const data = await response.json();
+      setAvailableRooms((prev) => {
+        const isRoomAlreadyExists = prev.some((room) => room.roomId === data.roomId);
+        if (!isRoomAlreadyExists) {
+          return [...prev, data];
+        }
+        return prev;
+      });
+      setRoomName('');
+      setRoomId(data.roomId);
+      setErrorMessage('');
+    } catch (err) {
+      console.error("Failed to create room:", err);
+    }
+  };
+
+  useEffect(() => {
+    const fetchRooms = async () => {
+      try {
+        const response = await fetch('http://localhost:3005/rooms');
+        const data = await response.json();
+        setAvailableRooms(data);
+      } catch (error) {
+        console.error('Failed to fetch rooms:', error);
+      }
+    };
+
+    fetchRooms();
+  }, []);
+
+  useEffect(() => {
+    const fetchConnectionsCount = async () => {
+      try {
+        const response = await fetch('http://localhost:3005/connections-count');
+        const data = await response.json();
+        setConnectionsCount(data.connectionsCount);
+      } catch (error) {
+        console.error('Failed to fetch connections count:', error);
```

Comment:
Similar to the previous suggestion, consider setting an error message for the user when fetching connections count fails. This will help users understand the issue.